File: my-folder/3570-count-of-substrings-containing-every-vowel-and-k-consonants-i/solution.py
class Solution:
    def countOfSubstrings(self, s: str, k: int) -> int:
        vowels = set("aeiou")
        count = 0
        consonant_count = 0
        # Generate all substrings
        for i in range(len(s)):
            for j in range(i + 1, len(s) + 1):
                substring = s[i:j]
                if vowels.issubset(set(substring)):

                    # Count consonants in the substring
                    consonants = sum(
                        1 for char in substring if char.isalpha() and char not in vowels
                    )
                    if consonants==k:
                        count+=1
        return count

File: my-folder/3570-count-of-substrings-containing-every-vowel-and-k-consonants-i/solution.cpp
#include <string>
#include <unordered_map>
#include <unordered_set>

class Solution {
public:
    int countOfSubstrings(std::string s, int k) {
        std::unordered_set<char> vowels = {'a', 'e', 'i', 'o', 'u'};
        int count = 0;
        int n = s.length();

        for (int start = 0; start < n; ++start) {
            std::unordered_map<char, int> vowel_count;
            int consonant_count = 0;
            
            for (int end = start; end < n; ++end) {
                char char_end = s[end];

                if (vowels.count(char_end)) {
                    vowel_count[char_end]++;
                } else if (isalpha(char_end)) {
                    consonant_count++;
                }
                
                // Check if all vowels are present
                if (vowel_count.size() == vowels.size()) {
                    if (consonant_count == k) {
                        count++;
                    }
                }
            }
        }
        
        return count;
    }
};

